r/tests/testthat/helper-expectation.R

```diff
@@ -189,3 +187,109 @@ expect_vector_equal <- function(expr, # A vectorized R expression containing `in
     skip(paste(skip_msg, collpase = "\n"))
   }
 }
+
+expect_vector_equivalent <- function(expr, # A vectorized R expression containing `input` as its input
```

Comment:
I think it's worthwhile to have this new helper function, but it would be better to implement it by adding an optional argument `ignore_attributes = FALSE` to `expect_vector_equal()`, using that to control whether `expect_equivalent()` or `expect_equal()` is used, and then calling that function from this one? That would reduce the amount of copied code.

You'll need to use `!!` to unquote `expr` in the call to  `expect_vector_equivalent()` from here.

Reply:
(Please do push back if you think I'm mistaken about this or if I'm missing some important details!)

Reply:
Another argument for `expect_vector_equal(ignore_attributes = TRUE)` is that testthat 3e deprecates `expect_equivalent()` in favor of `expect_equal(ignore_attr = TRUE)`.  And if we can name the argument, I would vote for `ignore_attr` over `ignore_attributes` to match testthat (well, I guess technically it's waldo at that point).

Reply:
Agreed—best to name the attribute`ignore_attr` 👍

Reply:
Also heads up that `expect_vector_equal()` calls a function named `expect_vector()` but this is _not_ `testthat::expect_vector()`; it's just a simple function defined higher up in `helper-expectation.R`. That confused me initially.

Reply:
We can rename our `expect_vector` if that helps; I believe `expect_vector` is new in testthat (at least it's new to me).

Reply:
➕ for renaming our `expect_vector` to avoid any possible confusion.

In other places I've been bitten by these testing helper name overlaps, and each time it's happened it threw me for a loop trying to figure out what was going wrong until I realized it was a name collision in a helper.

Reply:
Have now updated the code here to implement `ignore_attr`.  What makes sense as a new name for `expect_vector`?  Maybe `expect_as_vector` or `expect_vector_match`?

Reply:
Gone with `expect_as_vector` for now!